devices/homeetec.js

```diff
@@ -0,0 +1,287 @@
+            tuya.exposes.switch().withEndpoint('l4'),
+            tuya.exposes.backlightModeOffOn(),
+        ],
+        endpoint: (device) => {
+            return {'l1': 1, 'l2': 1, 'l3': 1, 'l4': 1};
+        },
+        meta: {
+            multiEndpoint: true,
+            tuyaDatapoints: [
+                [1, 'state_l1', tuya.valueConverter.onOff],
+                [2, 'state_l2', tuya.valueConverter.onOff],
+                [3, 'state_l3', tuya.valueConverter.onOff],
+                [4, 'state_l4', tuya.valueConverter.onOff],
+                [7, 'backlight_mode', tuya.valueConverter.onOff],
+            ],
+        },
+    },
+    {
+        fingerprint: [
+            {modelID: 'TS0601', manufacturerName: '_TZE200_p6vz3wzt'},
+        ],
+        model: 'TS0601_cover_5',
+        vendor: 'TuYa',
+        description: 'Curtain / Blind switch',
+        fromZigbee: [tuya.fz.homeetec_cover],
+        toZigbee: [tuya.tz.homeetec_cover],
+        exposes: [
+            e.cover_position(),
+            exposes.enum(DP_KEYS.Calibration, ea.STATE_SET, ['START', 'END']).withDescription('Calibration'),
+            exposes.binary(DP_KEYS.Backlight, ea.STATE_SET, 'ON', 'OFF').withDescription('Backlight'),
+            exposes.enum(DP_KEYS.MotorSteering, ea.STATE_SET, ['FORWARD', 'BACKWARD']).withDescription('Motor Steering'),
+            exposes.binary(DP_KEYS.ChildLock, ea.STATE_SET, 'ON', 'OFF').withDescription('Child Lock'),
+        ],
+    },
+    {
+        fingerprint: [
+            {modelID: 'TS0601', manufacturerName: '_TZE200_jhkttplm'},
+        ],
+        model: 'TS0601_cover_with_1_switch',
+        vendor: 'TuYa',
+        description: 'Curtain / Blind switch with 1 Gang switch',
+        fromZigbee: [tuya.fz.homeetec_cover],
+        toZigbee: [tuya.tz.homeetec_cover],
+        exposes: [                
+            e.cover_position(),
+            exposes.binary(DP_KEYS.Switch, ea.STATE_SET, 'ON', 'OFF').withDescription('Switch On/Off'),
+            exposes.enum(DP_KEYS.Calibration, ea.STATE_SET, ['START', 'END']).withDescription('Calibration'),
+            exposes.binary(DP_KEYS.Backlight, ea.STATE_SET, 'ON', 'OFF').withDescription('Backlight'),
+            exposes.enum(DP_KEYS.MotorSteering, ea.STATE_SET, ['FORWARD', 'BACKWARD']).withDescription('Motor Steering'),
+            exposes.binary(DP_KEYS.ChildLock, ea.STATE_SET, 'ON', 'OFF').withDescription('Child Lock'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [COVER_DP[DP_KEYS.Switch], DP_KEYS.Switch, tuya.valueConverter.onOff],
+            ],
+        },
+    },
+    {
+        fingerprint: [
+            {modelID: 'TS0601', manufacturerName: '_TZE200_5nldle7w'},
+        ],
+        model: 'TS0601_cover_with_2_switch',
+        vendor: 'TuYa',
+        description: 'Curtain / Blind switch with 2 Gang switch',
+        options: [exposes.options.invert_cover()],
+        fromZigbee: [tuya.fz.homeetec_cover],
```

Comment:
As far as I can see the only thing that `homeetec_cover` does is providing invert support for the position. You can use `[2, 'position', tuya.valueConverter.coverPosition],` and switch back to:


```js
       fromZigbee: [tuya.fz.datapoints],
       toZigbee: [tuya.tz.datapoints],
``

After this `tuya.fz.homeetec_cover`/`tuya.tz.homeetec_cover` can be removed. Example: https://github.com/Koenkk/zigbee-herdsman-converters/blob/3bcfdb6d1d992a34c2bbb9814665165e0fe6aea3/devices/tuya.js#L2647

Reply:
amazing, combined it with the multi endpoints for the switches,
to make it work the fix I applied for the 4-gang switch within `tuya.tz.datapoints`:
```javascript
const convertedKey = meta.mapped.meta.multiEndpoint && (meta.endpoint_name || false) ? `${key}_${meta.endpoint_name}` : key;
```

TS0601_cover_with_2_switch:
```javascript
{
        fingerprint: [
            {modelID: 'TS0601', manufacturerName: '_TZE200_5nldle7w'},
        ],
        model: 'TS0601_cover_with_2_switch',
        vendor: 'TuYa',
        description: 'Curtain / Blind switch with 2 Gang switch',
        options: [exposes.options.invert_cover()],
        fromZigbee: [tuya.fz.datapoints],
        toZigbee: [tzDataPoints],
        exposes: [                
            e.cover_position(),
            tuya.exposes.switch().withEndpoint('l1'),
            tuya.exposes.switch().withEndpoint('l2'),
            exposes.enum('calibration', ea.STATE_SET, ['START', 'END']).withDescription('Calibration'),
            exposes.binary('backlight_mode', ea.STATE_SET, 'ON', 'OFF').withDescription('Backlight'),
            exposes.enum('motor_steering', ea.STATE_SET, ['FORWARD', 'BACKWARD']).withDescription('Motor Steering'),
            exposes.binary('child_lock', ea.STATE_SET, 'ON', 'OFF').withDescription('Child Lock'),
        ],
        endpoint: (device) => {
            return {'l1': 1, 'l2': 1};
        },
        meta: {
            multiEndpoint: true,
            tuyaDatapoints: [
                [1, 'state', tuya.valueConverterBasic.lookup({'OPEN': new Enum(0), 'STOP': new Enum(1), 'CLOSE': new Enum(2)})],
                [2, 'position', tuya.valueConverter.coverPosition],
                [3, 'calibration', tuya.valueConverterBasic.lookup({'START': new Enum(0), 'END': new Enum(1)})],
                [7, 'backlight_mode', tuya.valueConverter.onOff],
                [8, 'motor_steering', tuya.valueConverterBasic.lookup({'FORWARD': new Enum(0), 'BACKWARDBACKWARD': new Enum(1)})],
                [101, 'state_l2', tuya.valueConverter.onOff],
                [102, 'state_l1', tuya.valueConverter.onOff],                    
                [103, 'child_lock', tuya.valueConverter.onOff],
            ],
        }
    }
```

will align it for the other covers by end of the week under `tuya.js`
(will also replace `new Enum(0)` to `tuya.enum(0)`)

Reply:
> amazing, combined it with the multi endpoints for the switches, to make it work the fix I applied for the 4-gang switch within tuya.tz.datapoints:

Why is this needed? (what doesn't work when not using that). There are other multi endpoint TuYa devices which work fine.

Reply:
Instead of having state, switch 1 &2, it creates state, state 1 &2 which is the standard (switch was not)